plugins/home/src/extensions.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { lazy, Suspense } from 'react';
+import { IconButton } from '@material-ui/core';
+import SettingsIcon from '@material-ui/icons/Settings';
+import { InfoCard } from '@backstage/core-components';
+import { SettingsModal } from './components';
+import { createReactExtension, useApp } from '@backstage/core-plugin-api';
+
+const lazyLoadedComponent = (
+  component: () => Promise<(props: any) => JSX.Element>,
+) => lazy(() => component().then(inner => ({ default: inner })));
+
+export function createHomePageComponentExtension({
+  title,
+  component,
+  actions,
+  contextProvider,
+  settings,
+}: {
+  title: string;
```

Comment:
Related to the `placeholder` example I gave below... As I was playing with the API, I found it a little limiting (from the perspective of an **App Integrator**) that I couldn't customize the card title.  I wonder if it's more a concern for the App Integrator than the Plugin Developer.

A `noCard` option (and thus, no title) would have also worked for me in this case, but I imagine App Integrators might generally want control over the card titles.

Reply:
Is it even the case that we want

- plugin exposes extension E1 with props a, b, c
- integrator "instantiates" or specializes the extension twice by setting props b and c to different values and and call them E2, E3
- what gets placed in the home page are some instances of E2 and E3, with different prop values for a

@benjdlambert @rubenLindstrom too

Reply:
Yes, I totally agree. This ties in to my reply to @benjdlambert above. If you create a renderer such as a `<HomePageComponentCard component={...} />`, it could naturally support overriding for instance whatever `title` is provided by the component developer. The end user could also interact with some of these things through the settings, if that is supported by the component developer.

Reply:
In 0cce460 this has been implemented so that component developers can register what custom props they will accept in a generic when calling `createCardExtension`.